packages/big-design/src/components/Radio/Radio.tsx

```diff
@@ -15,70 +16,64 @@ interface PrivateProps {
-        <HiddenRadio
-          type="radio"
-          checked={checked}
-          id={id}
-          disabled={disabled}
-          {...props}
-          aria-labelledby={this.labelUniqueId}
-          ref={forwardedRef}
-        />
-        <StyledRadio checked={checked} disabled={disabled} htmlFor={id} aria-hidden={true} />
-        {this.renderLabel()}
-      </RadioContainer>
-    );
-  }
-
-  private getInputId() {
-    const { id } = this.props;
-
-    return id ? id : this.uniqueId;
-  }
-
-  private renderLabel() {
-    const htmlFor = this.getInputId();
-    const { disabled, label } = this.props;
-
+const RawRadio: React.FC<RadioProps & PrivateProps> = ({
+  checked,
+  className,
+  disabled,
+  label,
+  forwardedRef,
+  style,
+  ...props
+}) => {
+  const id = useMemo(() => (props.id ? props.id : uniqueId('radio_')), [props.id]);
+  const labelId = useMemo(() => uniqueId('checkBox_label_'), []);
+
+  const renderedLabel = useMemo(() => {
     if (typeof label === 'string') {
       return (
-        <StyledLabel htmlFor={htmlFor} id={this.labelUniqueId} disabled={disabled} aria-hidden={disabled}>
+        <RadioLabel htmlFor={id} id={labelId} disabled={disabled} aria-hidden={disabled}>
           {label}
-        </StyledLabel>
+        </RadioLabel>
       );
     }
 
-    if (React.isValidElement(label) && label.type === Radio.Label) {
+    if (React.isValidElement(label) && label.type === RadioLabel) {
       return React.cloneElement(label as React.ReactElement<React.LabelHTMLAttributes<HTMLLabelElement>>, {
-        htmlFor,
-        id: this.labelUniqueId,
+        htmlFor: id,
+        id: labelId,
       });
     }
 
-    return null;
-  }
-}
-
-const RadioWithForwardedRef = React.forwardRef<HTMLInputElement, RadioProps>(({ className, style, ...props }, ref) => (
-  <RawRadio {...props} forwardedRef={ref} />
-));
-
-export const Radio = hoistNonReactStatics(RadioWithForwardedRef, RawRadio);
-
-export const StyleableRadio = React.forwardRef<HTMLInputElement, RadioProps>((props, ref) => (
-  <RawRadio {...props} forwardedRef={ref} />
-));
+    if (!label) {
+      return null;
+    }
 
-Radio.displayName = 'Radio';
-StyleableRadio.displayName = 'StyleableRadio';
+    warning('label must be either a string or a RadioLabel component.');
+  }, [disabled, id, label, labelId]);
+
+  return (
+    <RadioContainer className={className} style={style}>
+      <HiddenRadio
+        type="radio"
+        checked={checked}
+        id={id}
+        disabled={disabled}
+        {...props}
+        aria-labelledby={labelId}
+        ref={forwardedRef}
+      />
+      <StyledRadio checked={checked} disabled={disabled} htmlFor={id} aria-hidden={true} />
+      {renderedLabel}
+    </RadioContainer>
+  );
+};
+
+export const Radio = typedMemo(
+  forwardRef<HTMLInputElement, RadioProps>(({ className, style, ...props }, ref) => (
+    <RawRadio {...props} forwardedRef={ref} />
+  )),
+);
+
+export const StyleableRadio = typedMemo(
```

Comment:
Same as Checkbox, not part of your changes but probably worth it to remove the `StyleableRadio` export.